CodeChat_Server/codechat.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Installs CodeChat
+# test
+# To download this script and automatically run it, run in a terminal (without the # or $ signs):
+# $ wget https://raw.githubusercontent.com/bjones1/CodeChat_system/master/CodeChat_Server/codechat.sh && bash codechat.sh
+
+# check for Python 3
+if ! hash python3; then
+    echo "Python 3 is not installed"
+    exit 1
+fi
+
+# check python version
+python3 -c 'import sys'
+
+if ! python3 -c 'import sys; assert sys.version_info >= (3,7)' 2> /dev/null; then
+    echo "Upgrade Python to version 3.7 or above"
+    exit 1
+fi
+
+# install / upgrade pip
+python3 -m pip install --user --upgrade pip
+
+# create virtual environment
```

Comment:
Is venv installed by default? I think that ubuntu requires installing `python3-venv` or something like that. Perhaps it was `pip` that's not installed by default? In any case, please test on common Unix flavors (Debian/Ubuntu/Fedora/etc. -- perhaps https://www.tecmint.com/top-most-popular-linux-distributions/)?